src/algebra/algebra/basic.lean

```diff
@@ -851,6 +851,37 @@ ext $ λ x, show e₁.to_linear_map x = e₂.to_linear_map x, by rw H
 @[simp] lemma trans_to_linear_map (f : A₁ ≃ₐ[R] A₂) (g : A₂ ≃ₐ[R] A₃) :
   (f.trans g).to_linear_map = g.to_linear_map.comp f.to_linear_map := rfl
 
+section of_linear_equiv
+
+variables (l : A₁ ≃ₗ[R] A₂)
+  (map_mul : ∀ x y : A₁, l (x * y) = l x * l y)
+  (commutes : ∀ r : R, l (algebra_map R A₁ r) = algebra_map R A₂ r)
+
+/--
+Upgrade a linear equivalence to an algebra equivalence,
+given that it distributes over multiplication and action of scalars.
+-/
+def of_linear_equiv : A₁ ≃ₐ[R] A₂ :=
+{ to_fun := l,
+  inv_fun := l.symm,
+  left_inv := l.left_inv,
+  right_inv := l.right_inv,
+  map_mul' := map_mul,
+  map_add' := l.map_add',
+  commutes' := commutes }
+
+@[simp] lemma of_linear_equiv_to_linear_equiv :
+  of_linear_equiv (e.to_linear_equiv) e.map_mul e.commutes = e :=
+by { ext, refl }
```

Comment:
You want this apply to all proofes, not just ones from `e`:
```suggestion
@[simp] lemma of_linear_equiv_to_linear_equiv (map_mul) (commutes) :
  of_linear_equiv e.to_linear_equiv map_mul commutes = e :=
by { ext, refl }
```
(while lean is proof-irrelevant for defeq purposes, its not for the purpose of matching with `simp`)

I wonder though if the simp lemma we actually want is:
```
{ to_fun := e, inv_fun := e.symm, map_mul' := map_mul, commutes' := commutes, <etc>}.to_linear_equiv = e
```

Reply:
I've added some related lemmas in #6505, but not the one I mention above. It feels like there's a quadratic number of these lemmas, which is a bit unfortunate.

Reply:
I am not sure these will actually come up in practice any time soon. If, at that point, simp does not fire, we can add them then.